compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -255,10 +255,42 @@ private BPackageSymbol definePackage(DataInputStream dataInStream, int pkgCpInde
         // Define service declarations
         defineSymbols(dataInStream, rethrow(this::defineServiceDeclarations));
 
+        populateReferencedFunctions();
+
         this.typeReader = null;
         return this.env.pkgSymbol;
     }
 
+    private void populateReferencedFunctions() {
```

Comment:
Have we added tests for this change? Shall we add both positive and negative bala tests?

Reply:
This is covered under object_langlib_function_invocation_negative.bal

Reply:
That doesn't test the BIR symbol enter, right? We need to add a bala test?

Reply:
Added tests in 1b6b21ff9a8ad716b114fd76ad04f19dc70a57e8